File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/shop/client/clientResponse.ts
import { Client } from "../../../..";

export interface ClientResponse {
    id: string;
    userId: string;
    name: string;
    middleName: string;
    lastName: string;
    dateOfBirth: Date;
    address: string;
    phone: string;
    email: string;
}
export function mapClientResponseToClient(response: ClientResponse): Client {
    return {
        id: response.id,
        userId: response.userId,
        name: response.name,
        middleName: response.middleName,
        lastName: response.lastName,
        dateOfBirth: new Date(response.dateOfBirth),
        address: response.address,
        phone: response.phone,
        email: response.email,
    };
}

File: src/ELibrary.Backend/LibraryShopEntities/Domain/Dtos/Shop/ClientResponse.cs
namespace LibraryShopEntities.Domain.Dtos.Shop
{
    public class ClientResponse
    {
        public string? Id { get; set; }
        public string? UserId { get; set; }
        public string? Name { get; set; }
        public string? MiddleName { get; set; }
        public string? LastName { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string? Address { get; set; }
        public string? Phone { get; set; }
        public string? Email { get; set; }
    }
}